pkg/appgw/requestroutingrules.go

```diff
@@ -125,52 +137,42 @@ func (c *appGwConfigBuilder) RequestRoutingRules(cbCtx *ConfigBuilderContext) er
 		}
 	}
 
-	var urlPathMapFiltered []n.ApplicationGatewayURLPathMap
+	return urlPathMaps
+}
+
+func (c *appGwConfigBuilder) getRules(cbCtx *ConfigBuilderContext) ([]n.ApplicationGatewayRequestRoutingRule, []n.ApplicationGatewayURLPathMap) {
+	_, httpListenersMap := c.getListeners(cbCtx)
+	var pathMap []n.ApplicationGatewayURLPathMap
 	var requestRoutingRules []n.ApplicationGatewayRequestRoutingRule
-	for listenerID, urlPathMap := range urlPathMaps {
+	for listenerID, urlPathMap := range c.getURLPathMaps(cbCtx) {
 		httpListener := httpListenersMap[listenerID]
+		rule := n.ApplicationGatewayRequestRoutingRule{
+			Etag: to.StringPtr("*"),
+			Name: to.StringPtr(generateRequestRoutingRuleName(listenerID)),
+			ApplicationGatewayRequestRoutingRulePropertiesFormat: &n.ApplicationGatewayRequestRoutingRulePropertiesFormat{
+				HTTPListener: &n.SubResource{ID: to.StringPtr(c.appGwIdentifier.listenerID(*httpListener.Name))},
+			},
+		}
 		if len(*urlPathMap.PathRules) == 0 {
 			// Basic Rule, because we have no path-based rule
-			rule := n.ApplicationGatewayRequestRoutingRule{
-				Etag: to.StringPtr("*"),
-				Name: to.StringPtr(generateRequestRoutingRuleName(listenerID)),
-				ApplicationGatewayRequestRoutingRulePropertiesFormat: &n.ApplicationGatewayRequestRoutingRulePropertiesFormat{
-					RuleType:              n.Basic,
-					HTTPListener:          &n.SubResource{ID: to.StringPtr(c.appGwIdentifier.listenerID(*httpListener.Name))},
-					RedirectConfiguration: urlPathMap.DefaultRedirectConfiguration,
-				},
-			}
+			rule.RuleType = n.Basic
+			rule.RedirectConfiguration = urlPathMap.DefaultRedirectConfiguration
 
 			// We setup the default backend address pools and default backend HTTP settings only if
 			// this rule does not have an `ssl-redirect` configuration.
 			if rule.RedirectConfiguration == nil {
 				rule.BackendAddressPool = urlPathMap.DefaultBackendAddressPool
 				rule.BackendHTTPSettings = urlPathMap.DefaultBackendHTTPSettings
 			}
-			requestRoutingRules = append(requestRoutingRules, rule)
```

Comment:
This line is repeated and could be removed if the `rule` object exists in the outer scope.